### ISSUE TITLE
Difference between this and CuyZ/Valinor

### ISSUE THREAD

**Issue by loevgaard** [brick/json-mapper#2]
Hi, Ben

I have used your libraries frequently in the past. I just saw this library. What is the difference between this library and https://github.com/CuyZ/Valinor?

That's something you might add to the README because CuyZ/Valinor has gained quite a traction I reckon.

Thanks for your work :)

**Comment by BenMorel** [comment]
Hi, thanks for the pointer! I've never come across CuyZ/Valinor, I've tried / read the docs of a couple ones before writing brick/json-mapper, and wasn't happy with the results.

I've provided a quick comparison with what I thought were the two most advanced libraries on the topic in [a reddit thread](https://www.reddit.com/r/PHP/comments/113dt5s/comment/j8pkk2a/?utm_source=share&utm_medium=web2x&context=3) I started, and indeed someone mentioned Valinor as a response, among several others.

Honestly Valinor seems extremely advanced, and supports many features and type annotations that brick/json-mapper doesn't support (and, for some of them, may never support as it wasn't the original goal of this library).

I took a moment to read their docs, and it may check all the boxes of what I was looking for when I decided to create this library. 

One thing I'm unsure of is how they handle unions of DTO classes (brick/json-mapper would [automatically detect](https://github.com/brick/json-mapper#union-types) which class the JSON object matches) and complex types involving unions.

Other than that, apart from the slightly simpler API and a couple configuration options, Valinor may very well do everything brick/json-mapper does, and more.
